src/components/fields/SchemaField.js

```diff
@@ -164,7 +166,12 @@ function SchemaFieldRender(props) {
     formContext,
     FieldTemplate = DefaultTemplate,
   } = registry;
+  let idSchema = props.idSchema;
   const schema = retrieveSchema(props.schema, definitions, formData);
+  idSchema = mergeObjects(
+    toIdSchema(schema, null, definitions, formData, idPrefix),
+    idSchema
+  );
```

Comment:
Why do we need to generate another `idSchema` at this level? What's wrong with the one that was generated for us at toplevel? That one was generated from the resolved schema, so it should have the `oneOf` accounted for, shouldn't it?

I'm a little concerned that we're restarting with `id=null` here -- won't this lose our place in the hierarchy?

Reply:
Same question here, removing this piece of code doesn't seem to break any test.